runtime/default.policy

```diff
@@ -1,82 +1,123 @@
+#################################################################
+macros:
+  - id: DD_AGENT_PROCESSES
+    description: Processes that are a part of the Datadog Agent
+    expression: >-
+      ["/opt/datadog-agent/embedded/bin/agent", "/opt/datadog-agent/embedded/bin/system-probe", "/opt/datadog-agent/embedded/bin/security-agent", "/opt/datadog-agent/embedded/bin/process-agent"]
+
+  - id: CONTAINER_PROCESSES
+    description: Processes related to operating containers and kubernetes clusters
+    expression: >-
+      ["/usr/bin/containerd", "/usr/bin/docker", "/usr/bin/dockerd", "/usr/bin/docker-compose", "/usr/bin/kubelet", "/usr/bin/skydns", "exechealthz", "/usr/local/bin/weave-net", "/opt/cni/bin/loopback", "/opt/cni/bin/bridge", "openshift-sdn", "openshift"]
+      # TODO: hyperkube, kube2sky, exechealthz, openshift-sdn, openshift
+
+  - id: CREDENTIAL_PROCESSES
+    desription: Processes that access credential files as part of normal system activity
+    expression: >-
+      ["/usr/lib/accountsservice/accounts-daemon", "/usr/bin/login", "/usr/bin/newgrp", "/usr/bin/sg", "/usr/bin/shadowconfig", "/usr/bin/chage", "/usr/bin/chfn", "/usr/bin/chsh", "/usr/sbin/cron", "/usr/bin/expiry", "/usr/bin/gpasswd", "/usr/bin/passwd", "/usr/sbin/chgpasswd", "/usr/sbin/chpasswd", "/usr/sbin/cpgr", "/usr/sbin/cppw", "/usr/sbin/groupadd", "/usr/sbin/groupdel", "/usr/sbin/groupmems", "/usr/sbin/groupmod", "/usr/sbin/grpck", "/usr/sbin/grpconv", "/usr/sbin/grpunconv", "/usr/sbin/newusers", "/usr/sbin/pwck", "/usr/sbin/pwconv", "/usr/sbin/pwunconv", "/usr/bin/sudo", "/usr/sbin/useradd", "/usr/sbin/userdel", "/usr/sbin/usermod", "/usr/sbin/vigr", "/usr/sbin/vipw"]
+
+# Add allow-listed process filenames to this list. Uncomment to use.
+# Add to rules in order to apply this allow list.
+#  - id: ALLOWED_PROCESSES
+#    expression: >-
+#      process.filename not in ["/usr/bin/example"]
+
+################################################################
+# Rules determine what the Datadog security-agent will monitor.#
+################################################################
 rules:
   - id: credential_accessed
     description: Sensitive credential files were accessed using a non-standard tool
     expression: >-
       (open.filename == "/etc/shadow" || open.filename == "/etc/gshadow") &&
-      process.name not in ["vipw", "vigr", "accounts-daemon", "sudo", "cron"]
-  - id: memory_dump
-    description: Potential memory dump
-    expression: >-
-      open.filename =~ "/proc/*" && open.basename in ["maps", "mem"]
+      process.filename not in CREDENTIAL_PROCESSES
+
   - id: logs_altered
     description: Log data was deleted
     expression: >-
-      (open.filename =~ "/var/log/*" && open.flags & O_TRUNC > 0) &&
-      process.name not in ["agent", "security-agent", "process-agent", "system-probe", "kubelet", "containerd"]
+      (open.filename =~ "/var/log/*" && open.flags & O_TRUNC > 0)
```

Comment:
My understanding is that the 'truth table' of the Datadog part is as follows:
Datadog file            Process is Datadog          Alert
Yes                          Yes                                      No                         
Yes                           No                                      No
No                            Yes                                     No
No                            No                                      Yes

Are we sure we don't want to alert if a datadog agent truncates files outside of /var/log/datadog/, or if another process tries to truncate files inside of /var/log/datadog/? My guess is we want `!(open.filename !~ "/var/log/datadog/*" && process.filename not in DD_AGENT_PROCESSES)` (more visible, but unsure if it's possible) or `open.filename !~ "/var/log/datadog/*" || process.filename not in DD_AGENT_PROCESSES)`

Reply:
@safchain I'm unsure which of these options will work with the lexer, but Tanguy makes a good point. We know that the dd-agent should be allowed to truncate (ex: rotate) its own logs, but we should expose its actions to other logs (afaik, the agent will only ever read other logs, not truncate/rotate them)